src/neuroglancer/datasource/dvid/backend.ts

```diff
@@ -42,6 +44,38 @@ import {registerSharedObject} from 'neuroglancer/worker_rpc';
   }
 }
 
+export function decodeFragmentChunk(chunk: FragmentChunk, response: ArrayBuffer) {
+  let dv = new DataView(response);
+  let numVertices = dv.getUint32(0, true);
+  decodeTriangleVertexPositionsAndIndices(
+      chunk, response, Endianness.LITTLE, /*vertexByteOffset=*/4, numVertices);
+}
+
+@registerSharedObject() export class DVIDMeshSource extends
+(WithParameters(MeshSource, MeshSourceParameters)) {
+  download(chunk: ManifestChunk) {
+    // DVID does not currently store meshes chunked, the main
+    // use-case is for low-resolution 3D views.
+    // for now, fragmentId is the body id
+    return new Promise(function(resolve) {
+             resolve();
+           })
+        .then(() => {
+          chunk.fragmentIds = [`${chunk.objectId}`];
```

Comment:
You can simplify this code by just setting chunk.fragmentIds = ... and returning Promise.resolve(undefined);  no need for new Promise.

Reply:
I was wondering the most idiomatic way to do this.  Thanks for the feedback.  I will post a change shortly.